libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Views/SFSDKWindowManager.m

```diff
@@ -0,0 +1,304 @@
+    SFSDKWindowContainer *container = [[SFSDKWindowContainer alloc] initWithWindow:window andName:kSFMainWindowKey];
+    
+    container.windowType = SFSDKWindowTypeMain;
+    [container addDelegate:self];
+    _prevActiveWindow = container;
+    _currentWindow = container;
+    [self.namedWindows setObject:container forKey:kSFMainWindowKey];
+}
+
+- (SFSDKWindowContainer *)authWindow {
+    SFSDKWindowContainer *window = [self.namedWindows objectForKey:kSFLoginWindowKey];
+    if (!window) {
+        window = [self createAuthWindow];
+    }
+    return window;
+}
+
+- (SFSDKWindowContainer *)snapshotWindow {
+    SFSDKWindowContainer *window = [self.namedWindows objectForKey:kSFSnaphotWindowKey];
+    if (!window) {
+        window = [self createSnapshotWindow];
+    }
+    return window;
+}
+
+- (SFSDKWindowContainer *)passcodeWindow {
+    SFSDKWindowContainer *window = [self.namedWindows objectForKey:kSFPasscodeWindowKey];
+    if (!window) {
+        window = [self createPasscodeWindow];
+    }
+    return window;
+}
+
+- (SFSDKWindowContainer *)createNewNamedWindow:(NSString *)windowName {
+    SFSDKWindowContainer * container = nil;
+    if ( ![self isReservedName:windowName] ) {
+        UIWindow *window = [self createDefaultUIWindow];
+        container = [[SFSDKWindowContainer alloc] initWithWindow:window andName:windowName];
+        container.windowType = SFSDKWindowTypeOther;
+        [container addDelegate:self];
+        [self.namedWindows setObject:container forKey:windowName];
+    }
+    return container;
+}
+
+- (BOOL)isReservedName:(NSString *) windowName {
+    return ([windowName isEqualToString:kSFMainWindowKey] ||
+            [windowName isEqualToString:kSFLoginWindowKey] ||
+            [windowName isEqualToString:kSFPasscodeWindowKey] ||
+            [windowName isEqualToString:kSFSnaphotWindowKey]);
+    
+}
+
+- (BOOL)removeNamedWindow:(NSString *)windowName {
+    BOOL result = NO;
+    if (![self isReservedName:windowName]) {
+        [self.namedWindows removeObjectForKey:windowName];
+        result = YES;
+    }
+    return result;
+}
+
+- (SFSDKWindowContainer *)windowWithName:(NSString *)name {
+    return [self.namedWindows objectForKey:name];
+}
+
+- (void)bringToFront:(SFSDKWindowContainer *)windowContainer {
+    
+    if (!windowContainer)
+        return;
+
+    SFSDKWindowContainer *snapshotWindow = [self.namedWindows objectForKey:kSFSnaphotWindowKey];
+    
+    if ((_currentWindow != snapshotWindow) && (_currentWindow!=windowContainer) )
+        _prevActiveWindow = _currentWindow;
+    
+    _currentWindow = windowContainer;
+    [windowContainer makeKeyVisible];
+    
+    //send the rest of the windows to back
+    for ( NSString *wName in self.namedWindows.keyEnumerator ) {
+        SFSDKWindowContainer *wContainer = [self.namedWindows objectForKey:wName];
+        if ( ![wName isEqualToString:windowContainer.windowName] ) {
+            [wContainer sendToBack];
```

Comment:
@khawkins Merely setting distinct window levels doesn't work as intended. If the send2back is not done (set the other window levels to lower), I've observed that the current key window  becomes unresponsive although it is the first responder & is visible. Comment this block of code you will notice the effects.